src/core/mainloop/mainloop.c

```diff
@@ -1382,16 +1382,27 @@ CALLBACK(write_stats_file);
 STATIC periodic_event_item_t periodic_events[] = {
   /* Everyone needs to run those. */
   CALLBACK(add_entropy, ALL, 0),
-  CALLBACK(check_expired_networkstatus, ALL, 0),
-  CALLBACK(clean_caches, ALL, 0),
-  CALLBACK(fetch_networkstatus, ALL, 0),
   CALLBACK(heartbeat, ALL, 0),
-  CALLBACK(launch_descriptor_fetches, ALL, FL(NEED_NET)),
-  CALLBACK(reset_padding_counts, ALL, 0),
+
+  /* XXXX Do we have a reason to do this on a callback? */
   CALLBACK(retry_listeners, ALL, FL(NEED_NET)),
-  CALLBACK(save_state, ALL, 0),
-  CALLBACK(rotate_x509_certificate, ALL, 0),
-  CALLBACK(write_stats_file, ALL, 0),
+
+  /* We need to do these if we're participating in the Tor network. */
+  CALLBACK(check_expired_networkstatus, NET_PARTICIPANT, 0),
+  CALLBACK(fetch_networkstatus, NET_PARTICIPANT, 0),
```

Comment:
If this becomes net participant then the `hs_service` callback needs also to become net participant because it would be useless wake up every second if we do not have a live consensus or even able to create circuits.

Reply:
So for now, having any hs service at all keeps you from becoming a non-participant -- see the second check in check_network_participation_callback().